pkg/controllers/accesslogpolicy_controller.go

```diff
@@ -103,18 +103,26 @@ func RegisterAccessLogPolicyController(
 }
 
 func (r *accessLogPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	r.log.Infow("reconcile", "name", req.Name)
+	ctx = gwlog.NewTrace(ctx)
+	gwlog.AddMetadata(ctx, "type", "accesslogpolicy")
+	gwlog.AddMetadata(ctx, "name", req.Name)
+
+	r.log.Infow(ctx, "reconcile starting", gwlog.GetMetadata(ctx)...)
+	defer func() {
+		r.log.Infow(ctx, "reconcile completed", gwlog.GetMetadata(ctx)...)
+	}()
```

Comment:
here's an example of how I attempted to implement somehting like https://stripe.com/blog/canonical-log-lines

at the end of the reconcile event we can print out all the metadata saved in the context so far. eg this can include information like if issues occurred during the reconcile attempt that were ignorable, or decisions made while servicing the event.